packages/vm/src/context.rs

```diff
@@ -19,31 +19,59 @@ use cosmwasm_std::KV;
 
 #[cfg(feature = "iterator")]
 use crate::errors::{make_iterator_does_not_exist, FfiResult};
-use crate::errors::{make_uninitialized_context_data, VmResult};
+use crate::errors::{make_uninitialized_context_data, VmError, VmResult};
+#[cfg(feature = "iterator")]
+use crate::traits::StorageIteratorItem;
 use crate::traits::{Querier, Storage};
 
 /** context data **/
 
+pub struct GasState {
+    /// Gas limit for the computation.
+    gas_limit: u64,
+    /// Tracking the gas used in the cosmos SDK, in cosmwasm units.
+    sdk_used_gas: u64,
```

Comment:
I think this is slightly misleading, since SDK gas is already consumed when the transaction signature is verified.

What about

```rust
    /// Tracking the gas used during calls into the backend (Cosmos SDK), in cosmwasm units.
    /// This increases during e.g. storage operations and API calls.
    backend_used_gas: u64,
```

Reply:
yeah, sure

Reply:
Alternative idea: `externally_used_gas`

Reply:
I like this more. `backend_used_gas` makes it sound like it's coming from the wasmer backend, where it is the opposite.